https://lore.kernel.org/git/371571630df201b0b0c680f6437decc6c60ecd3b.1736307757.git.sam@gentoo.org/

From 371571630df201b0b0c680f6437decc6c60ecd3b Mon Sep 17 00:00:00 2001
Message-ID: <371571630df201b0b0c680f6437decc6c60ecd3b.1736307813.git.sam@gentoo.org>
From: Sam James <sam@gentoo.org>
Date: Wed, 8 Jan 2025 03:40:55 +0000
Subject: [PATCH] meson: fix perl dependencies

`generate_perl_command` needs `depends: [git_version_file]` and the uses
in top-level meson.build were fine, but the ones in perl/ weren't, causing
parallel build failures in some cases as GIT-BUILD-OPTIONS wasn't yet
available.

Signed-off-by: Sam James <sam@gentoo.org>
--- a/perl/FromCPAN/Mail/meson.build
+++ b/perl/FromCPAN/Mail/meson.build
@@ -4,4 +4,5 @@ test_dependencies += custom_target(
   command: generate_perl_command,
   install: true,
   install_dir: get_option('datadir') / 'perl5/FromCPAN/Mail',
+  depends: [git_version_file],
 )
--- a/perl/FromCPAN/meson.build
+++ b/perl/FromCPAN/meson.build
@@ -4,6 +4,7 @@ test_dependencies += custom_target(
   command: generate_perl_command,
   install: true,
   install_dir: get_option('datadir') / 'perl5/FromCPAN',
+  depends: [git_version_file],
 )
 
 subdir('Mail')
--- a/perl/Git/LoadCPAN/Mail/meson.build
+++ b/perl/Git/LoadCPAN/Mail/meson.build
@@ -4,4 +4,5 @@ test_dependencies += custom_target(
   command: generate_perl_command,
   install: true,
   install_dir: get_option('datadir') / 'perl5/Git/LoadCPAN/Mail',
+  depends: [git_version_file],
 )
--- a/perl/Git/LoadCPAN/meson.build
+++ b/perl/Git/LoadCPAN/meson.build
@@ -4,6 +4,7 @@ test_dependencies += custom_target(
   command: generate_perl_command,
   install: true,
   install_dir: get_option('datadir') / 'perl5/Git/LoadCPAN',
+  depends: [git_version_file],
 )
 
 subdir('Mail')
--- a/perl/Git/SVN/Memoize/meson.build
+++ b/perl/Git/SVN/Memoize/meson.build
@@ -4,4 +4,5 @@ test_dependencies += custom_target(
   command: generate_perl_command,
   install: true,
   install_dir: get_option('datadir') / 'perl5/Git/SVN',
+  depends: [git_version_file],
 )
--- a/perl/Git/SVN/meson.build
+++ b/perl/Git/SVN/meson.build
@@ -14,6 +14,7 @@ foreach source : [
     command: generate_perl_command,
     install: true,
     install_dir: get_option('datadir') / 'perl5/Git/SVN',
+    depends: [git_version_file],
   )
 endforeach
 
--- a/perl/Git/meson.build
+++ b/perl/Git/meson.build
@@ -11,6 +11,7 @@ foreach source : [
     command: generate_perl_command,
     install: true,
     install_dir: get_option('datadir') / 'perl5/Git',
+    depends: [git_version_file],
   )
 endforeach
 
--- a/perl/meson.build
+++ b/perl/meson.build
@@ -4,6 +4,7 @@ test_dependencies += custom_target(
   command: generate_perl_command,
   install: true,
   install_dir: get_option('datadir') / 'perl5',
+  depends: [git_version_file],
 )
 
 subdir('Git')
-- 
2.48.0.rc2.29.g7bab57bd98

